src/networking.c

```diff
@@ -1911,7 +1911,9 @@ void processInputBuffer(client *c) {
     /* Keep processing while there is something in the input buffer */
     while(c->qb_pos < sdslen(c->querybuf)) {
         /* Return if clients are paused. */
-        if (!(c->flags & CLIENT_SLAVE) && clientsArePaused()) break;
+        if (!(c->flags & CLIENT_SLAVE) && 
+            !(c->flags & CLIENT_PENDING_READ) && 
```

Comment:
does this mean we risk violating the pause in order to avoid violating the thread safety?
maybe we cam drain the IO thread assigned clients before we start the pause?

Reply:
Technically this PR doesn't change that guarantee. However, the current code does allow mutative commands to be executed after pause.  It queues up fully formed commands during the fan out, and then executes them regardless of the client pause state since there is no pause check in processCommand(). Looks like pause never fundamentally worked with threaded reads, so we'll need to do some draining/re-queueing.